### PR TITLE
Clone subvolumes properly

### PR DESCRIPTION
**Pull request by imobachgs**
Now subvolumes are cloned in a proper way.

### CODE REVIEW

File: src/lib/y2storage/filesystems/btrfs.rb [hunk 1]
```diff
@@ -301,6 +301,18 @@ def restore_unshadowed_subvolumes(devicegraph)
...
       end
 
+      # Determines the default subvolume name
+      #
+      # When a default subvolume name have been used, a subvolume named after
+      # it lives under the #top_level_btrfs_subvolume. Otherwise, an empty
+      # string will be taken as the default subvolume name.
+      #
+      # @return [String] Default subvolume name
+      def default_subvolume_name
```
Comment (joseivanlopez) [inline src/lib/y2storage/filesystems/btrfs.rb]: I am not sure if this is the best name for the method. We have a `Btrfs#default_btrfs_subvolume` method, and `#default_subvolume_name` looks like the name (path) of that subvolume, but it is not. Probably we need a new subvolume concept, something like `top_parent_subvolume`. Something to discuss in future, I guess.

File: src/lib/y2storage/autoinst_profile/partition_section.rb [hunk 2]
```diff
@@ -288,6 +291,23 @@ def init_mount_options(fs)
         @fstab_options = fs.fstab_options unless fs.fstab_options.empty?
       end
 
+      # @param fs [Filesystem::BlkFilesystem]
```
Comment (joseivanlopez) [inline src/lib/y2storage/autoinst_profile/partition_section.rb]: It is obvious, but maybe to add a comment saying that it populates `@subvolumes` ¿?

File: test/y2storage/filesystems/btrfs_test.rb [hunk 3]
```diff
@@ -639,4 +639,22 @@
...
+    end
+
+    context "when default subvolume name is not set" do
+      let(:dev_name) { "/dev/sdd1" }
+
+      it "returns an empty string" do
+        expect(subject.default_subvolume_name).to eq("")
+      end
+    end
+  end
```
Comment (joseivanlopez) [inline test/y2storage/filesystems/btrfs_test.rb]: I would add one more tests where the `default_btrfs_subvolume.path` and `default_subvolume_name` are different, to clarify that they are not related. For example, with snapshots, the `default_btrfs_subvolume.path` would be `@/.snapshots`, isn't it?

File: src/lib/y2storage/filesystems/btrfs.rb (hunk 1 above)
Comment (imobachgs) [reply src/lib/y2storage/filesystems/btrfs.rb]: Yes, the name can be tricky. For instance, this concept is named `btrfs_default_subvolume` in the control file, which is even worst: https://github.com/yast/yast-storage-ng/blob/master/test/data/control_files/caasp.xml#L31

File: test/y2storage/filesystems/btrfs_test.rb (hunk 3 above)
Comment (imobachgs) [reply test/y2storage/filesystems/btrfs_test.rb]: It would be `@/.snapshots/1/snapshot` (or whatever snapshot it is the current one). And the `default_subvolume_name` could be `@`.

For instance, the second test checks that `default_subvolume_name` is not set and, although subvolumes are named like `@/blah`, the `default_subvolume_name` is still `""`.

File: src/lib/y2storage/filesystems/btrfs.rb (hunk 1 above)
Comment (imobachgs) [reply src/lib/y2storage/filesystems/btrfs.rb]: The problem when using `top` is that the `top_level` is a different thing and it could be confusing too.

**Comment by imobachgs** [comment]
As discussed with @joseivanlopez on IRC, I will hold the PR until we discuss about `default_subvolume_name` name.

**Comment by imobachgs** [comment]
After the discussion, I've changed that confusing name to `subvolumes_prefix`.

**Comment by imobachgs** [comment]
Thanks!
